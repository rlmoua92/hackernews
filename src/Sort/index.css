.button-active {
    border-radius: 0;
    border-bottom: 1px solid #38BB6C;
}