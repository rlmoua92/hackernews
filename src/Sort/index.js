import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './index.css';
import Button from '../Button/index.js';

const Sort = ({ 
  sortKey,
  activeSortKey, 
  onSort, 
  children,
  isSortReverse 
}) => {
  const sortClass = classNames(
    'button-inline',
    { 'button-active': sortKey === activeSortKey}
  );

  const sortIcon = 
    sortKey === activeSortKey
      ? isSortReverse
        ? <i className="fas fa-angle-down" style={{ paddingLeft: '5px' }}></i>
        : <i className="fas fa-angle-up" style={{ paddingLeft: '5px' }}></i>
      : null;

  return (
    <Button 
      onClick={() => onSort(sortKey)}
      className={sortClass}
    >
      {children}{sortIcon}
    </Button>
  );
};

Sort.propTypes = {
  sortKey: PropTypes.string.isRequired,
  activeSortKey: PropTypes.string.isRequired,
  onSort: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
  isSortReverse: PropTypes.bool.isRequired,
};

export default Sort;