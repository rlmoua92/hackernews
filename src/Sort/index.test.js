import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Sort from './index.js';

Enzyme.configure({ adapter: new Adapter() });

describe('Sort', () => {
  const props = {
    sortKey: 'TITLE',
    activeSortKey: 'TITLE',
    isSortReverse: false,
  }

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Sort {...props} onSort={() => console.log("SORT BY TITLE")}>Title</Sort>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <Sort {...props} onSort={() => console.log("SORT BY TITLE")}>Title</Sort>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});