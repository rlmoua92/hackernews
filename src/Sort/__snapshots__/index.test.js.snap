// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort has a valid snapshot 1`] = `
<button
  className="button-inline button-active"
  onClick={[Function]}
  type="button"
>
  Title
  <i
    className="fas fa-angle-up"
    style={
      Object {
        "paddingLeft": "5px",
      }
    }
  />
</button>
`;
